### ISSUE TITLE
ERROR - node:internal/modules/cjs/loader:936   throw err;

### ISSUE THREAD

**Issue by GabrielCHP** [nodejs/help#3709]
### Platform

Windows Powershell

### Subsystem

Powershell

### What steps will reproduce the bug?

Downloading node.js > Opening the Windows Powershell > Selecting the Folder > node install.  # Not working!

### What do you see instead?

node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\__MACOSX\hashlips\install'
←[90m    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)←[39m
←[90m    at Function.Module._load (node:internal/modules/cjs/loader:778:27)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)←[39m
←[90m    at node:internal/main/run_main_module:17:47←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: []
}

### Additional information

![BUG](https://user-images.githubusercontent.com/98562080/151458768-04480bff-46af-4820-96d4-19a6bdb5c771.png)

**Comment by Goldzik2** [comment]
I have the same problem

**Comment by Orkpo** [comment]
Same problem, any solution??

**Comment by Glitchbes** [comment]
same shit!

**Comment by Soroushj1** [comment]
same problem

**Comment by Glitchbes** [comment]
Figured it out. I left some things out in main.js

**Comment by abvillarreal** [comment]
Hello, the same issue
i even tried reinstalling everything, and nothing good happened.
I thought the problem was the version because i installed v17.5.0, so i uninstalled and installed v16.14.0
but same thing.
I manually tried to install the package with the command  `npm i npm-cli`  or command  `npm install npm-cli` 
The first time before downgrading node version, it was working fine the installation but errors were still there. after the downgrade, doesn't even try to install because immediately after shows the error.
Is there anybody who passed through this and is now happily ever after creating new projects..

BR everyone

**Comment by nkilm** [comment]
> 

Yeah I'm facing similar problem.
![image](https://user-images.githubusercontent.com/79012023/154729340-5e2663fd-60cd-4f4c-9753-e13ce5fe7a85.png)

Any solution(s) for this?

**Comment by AdamSioud** [comment]
I also need help here! much needed, thanks:)

**Comment by devpauliuss** [comment]
Same i need help, thanks for any information 👍

**Comment by nkilm** [comment]
Recently I was doing one simple project which would show Riddles in Command Line. I published it as npm package, but when I execute it (npx riddles-cli) I get the above mentioned error. 

Repository link:
`https://github.com/nkilm/riddles-in-CLI-a-npm-package`

npm package
`https://www.npmjs.com/package/riddles-cli`

**Comment by marsonya** [comment]
Can someone post a code snippet that can be used to reproduce the example? 
Kindly share details of the steps followed and commands. It will help in understanding and resolving the issue.

**Comment by ralfs66** [comment]
Following this tutorial results in specified error 
https://www.youtube.com/watch?v=KkZ6iYnSDRw

**Comment by nkilm** [comment]
any updates?

**Comment by horlathunbhosun** [comment]
Any updates on this, currently facing this too @

**Comment by nkilm** [comment]
####  One thing I observed is that this error occurs when we don't install the required dependencies correctly. But this works only for some cases, but it didn't work for me. 

So for now try this, 

- Delete the node_modules folder 
- Re-Install the necessary modules

Hope this helps😅

**Comment by dookpook** [comment]
yooo thanks man, im gonna try that rn. but how do i re-install the necessary modules?

**Comment by nkilm** [comment]
Welcome:)

All the type of dependencies are present in `package.json` file,

So all you have to do is, 
open your terminal in project folder  and execute the below command.

```bash
npm i
```
or

```bash
npm install 
```

Both should work fine😅

**Comment by dookpook** [comment]
its still giving this error 
![2022-03-09 17_57_11-Desktop](https://user-images.githubusercontent.com/101276441/157491544-f6aecf34-de8b-448b-aee1-b390d7ccda6d.png)
could it maybe be the code?

**Comment by dookpook** [comment]
could you add me on discord so we could think of a solution or so you could let me now if theres a solutions found?
its dookpook#0001

**Comment by taleleni** [comment]
Hello All,
I installed the very latest version 17.7.0 on windows but got the same error :( 
I even tried restalling and uninstalling too. but the same error coms with version 16 too. 


Is there anybody who passed through this 
. Kindly let me know.

**Comment by taleleni** [comment]
> Welcome:)
> 
> All the type of dependencies are present in `package.json` file,
> 
> So all you have to do is, open your terminal in project folder and execute the below command.
> 
> ```shell
> npm i
> ```
> 
> or
> 
> ```shell
> npm install 
> ```
> 
> Both should work fine😅

not working. getting same error

**Comment by dookpook** [comment]
> > Welcome:)
> > All the type of dependencies are present in `package.json` file,
> > So all you have to do is, open your terminal in project folder and execute the below command.
> > ```shell
> > npm i
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > or
> > ```shell
> > npm install 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Both should work fine😅
> 
> not working. getting same error

i know man had the same thing );

**Comment by Soroushj1** [comment]
I basically installed node again not through the commandline and it worked! I know it should work the same… but it didnt

**Comment by dookpook** [comment]
i was thinking of resetting my pc and starting over my project, is this gonna fix the issue?

**Comment by nkilm** [comment]
> could you add me on discord so we could think of a solution or so you could let me now if theres a solutions found? its dookpook#0001

Yeah sure I'll add you!

My discord username: nkilm#7651

**Comment by nkilm** [comment]
> Hello All, I installed the very latest version 17.7.0 on windows but got the same error :( I even tried restalling and uninstalling too. but the same error coms with version 16 too.
> 
> Is there anybody who passed through this . Kindly let me know.

Already tried that one, but no luck.

**Comment by nkilm** [comment]
> i was thinking of resetting my pc and starting over my project, is this gonna fix the issue?

I don't think it is related to PC, something with nodejs itself.

**Comment by dookpook** [comment]
> > i was thinking of resetting my pc and starting over my project, is this gonna fix the issue?
> 
> I don't think it is related to PC, something with nodejs itself.

alr accepted the fr

**Comment by nkilm** [comment]
> Welcome:)
> 
> All the type of dependencies are present in `package.json` file,
> 
> So all you have to do is, open your terminal in project folder and execute the below command.
> 
> ```shell
> npm i
> ```
> 
> or
> 
> ```shell
> npm install 
> ```
> 
> Both should work fine😅

One way to reproduce this error and actually fix:

Create one simple express application, but don't install express.
Just import and execute the code once. We'll get the above mentioned error.

Now to fix the issue install express using npm i express.

Now which gives us a hint that something is missing from project which is causing this error.

**Comment by dookpook** [comment]
maybe sounds dumb, how to actually install express? could u help me via discord maybe?

**Comment by itsme-vivek** [comment]
> maybe sounds dumb, how to actually install express? could u help me via discord maybe?

npm install express

**Comment by greatAznur** [comment]
i have same probleme  no soluton ?

**Comment by greatAznur** [comment]
[
](https://stackoverflow.com/questions/24721903/npm-npm-cli-js-not-found-when-running-npm)SET PATH=C:\Program Files\Nodejs;%PATH%
ok good

**Comment by nkilm** [comment]
> [stackoverflow.com/questions/24721903/npm-npm-cli-js-not-found-when-running-npm](https://stackoverflow.com/questions/24721903/npm-npm-cli-js-not-found-when-running-npm) SET PATH=C:\Program Files\Nodejs;%PATH% ok good

I had the same path, but still gives me this error:(

**Comment by jpilgrim** [comment]
I thought I had the same problem... and only after drinking a cup of coffee I noticed my mistake: Instead of `npm install` I had typed `node install`... and of course no such module could be found. From the initial error description this looks very similar.

**Comment by JeLoAL** [comment]
anyone solve this problem? all my node project giving me the same error

**Comment by BH88** [comment]
C:\Users\Bur\Desktop\We>npm i
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\npm-cli.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'C:\Program Files\nodejs\node_modules\npm\bin\node_modules\npm\bin\npm-cli.js'
←[90m    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)←[39m
←[90m    at Function.Module._load (node:internal/modules/cjs/loader:778:27)←[39m
←[90m    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)←[39m
←[90m    at node:internal/main/run_main_module:17:47←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: []
}

**Comment by jonasnapoles** [comment]
Unfortunately I have the same problem

**Comment by jonasnapoles** [comment]
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'semver'
Require stack:
- /usr/share/npm/lib/utils/unsupported.js
- /usr/share/npm/bin/npm-cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/usr/share/npm/lib/utils/unsupported.js:2:14)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/usr/share/npm/lib/utils/unsupported.js',
    '/usr/share/npm/bin/npm-cli.js'
  ]
}

**Comment by IamGideonIdoko** [comment]
I get a similar error when I run `yarn create react-app <app-name>`

```bash
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module 'ignore'
Require stack:
- C:\IFEXARCHIVE\IFEXPROJECT\graphql-app\frontend5\node_modules\globby\index.js
- C:\IFEXARCHIVE\IFEXPROJECT\graphql-app\frontend5\node_modules\react-scripts\scripts\utils\verifyTypeScriptSetup.js
- C:\IFEXARCHIVE\IFEXPROJECT\graphql-app\frontend5\[eval]
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\IFEXARCHIVE\IFEXPROJECT\graphql-app\frontend5\node_modules\globby\gitignore.js:6:19)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\node_modules\\globby\\gitignore.js',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\node_modules\\globby\\index.js',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\node_modules\\react-dev-utils\\globby.js',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\node_modules\\react-scripts\\scripts\\utils\\verifyTypeScriptSetup.js',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\node_modules\\react-scripts\\scripts\\init.js',
    'C:\\IFEXARCHIVE\\IFEXPROJECT\\graphql-app\\frontend5\\[eval]'
  ]
}
```

How can I resolve this?

**Comment by jonasnapoles** [comment]
> 

I recommend reinstalling or updating the version

**Comment by IamGideonIdoko** [comment]
> > 
> 
> I recommend reinstalling or updating the version



> > 
> 
> I recommend reinstalling or updating the version
> 
> Example
> 
> > node -v
> > n stable
> 
> Ubuntu
> 
> > sudo npm install n -g
> > sudo n stable

I'm currently using the latest stable version of yarn (v1.22.18) and node.js lts (v16.14.2). I still get the error whenever I use `yarn` to scaffold a React application. Although, when I use `npx` to generate the react app, it works fine.

I want to use `yarn` to manage my project's dependencies.

**Comment by IamGideonIdoko** [comment]
Ok. I just fixed it by:
* adding ignore `"ignore": "^5.2.0"` in my `package.json` as a dependency
* deleteting `node_modules` folder
*  running `yarn install`

**Comment by jeremiasmarinho** [comment]
On Windows I solved it like this.

I went to the folder: C:\Users\Documents\AppData\Roaming\npm

And I deleted the file. create-react-app.

I started the project again, and it worked.

**Comment by greatAznur** [comment]
on window i have  repair  the nodejs with  window  add and sup programe   shearch node js  click to modifie and repair

**Comment by jonasnapoles** [comment]
> Ok. I just fixed it by:
> 
> * adding ignore `"ignore": "^5.2.0"` in my `package.json` as a dependency
> * deleteting `node_modules` folder
> * running `yarn install`

Very good! It's very similar to all the cases before!

**Comment by celestinecyr** [comment]
@jonasnapoles do you also have a `node_modules` in your root?
Could you please share the structure of your project for comparison? Thank you!

**Comment by jonasnapoles** [comment]
![image](https://user-images.githubusercontent.com/81438449/160482041-fc813fd0-5acd-4107-a7fc-5df573d48e43.png)

**Comment by jonasnapoles** [comment]
> @jonasnapoles do you also have a `node_modules` in your root? Could you please share the structure of your project for comparison? Thank you!

@celestinecyr  This?

**Comment by celestinecyr** [comment]
Hi @jonasnapoles , I meant, do you have a `node_modules` folder outside of your `client` and `server` folder?

It's working perfectly fine when i try running `heroku local`, 
<img width="449" alt="Screenshot 2022-03-29 at 9 58 00 AM" src="https://user-images.githubusercontent.com/72538235/160517166-5b0049bf-981e-48b1-acad-af519ffbac5f.png">
and I'm able to build after having a `package.json` outside of my `client` and `server` file:
<img width="274" alt="Screenshot 2022-03-29 at 9 59 39 AM" src="https://user-images.githubusercontent.com/72538235/160517367-eb4fd089-6f8e-48aa-a21a-3c7d1d969f6e.png">

However it's still Application Error on Heroku even after it builds successfully. Do you have any idea why?

**Comment by jonasnapoles** [comment]
> Hi @jonasnapoles , I meant, do you have a `node_modules` folder outside of your `client` and `server` folder?
> 
> It's working perfectly fine when i try running `heroku local`, <img alt="Screenshot 2022-03-29 at 9 58 00 AM" width="449" src="https://user-images.githubusercontent.com/72538235/160517166-5b0049bf-981e-48b1-acad-af519ffbac5f.png">
> 
> and I'm able to deploy on heroku too after having a `package.json` outside of my `client` and `server` file: <img alt="Screenshot 2022-03-29 at 9 59 39 AM" width="274" src="https://user-images.githubusercontent.com/72538235/160517367-eb4fd089-6f8e-48aa-a21a-3c7d1d969f6e.png">
> 
> However it's still Application Error on Heroku even after it builds successfully. Do you have any idea why?
> 
> **My error:** node:internal/modules/cjs/loader:936 2022-03-29T01:47:57.534626+00:00 app[web.1]: throw err; 2022-03-29T01:47:57.534626+00:00 app[web.1]: ^ 2022-03-29T01:47:57.534627+00:00 app[web.1]: 2022-03-29T01:47:57.534627+00:00 app[web.1]: Error: Cannot find module '/app/index.js' 2022-03-29T01:47:57.534628+00:00 app[web.1]: at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15) 2022-03-29T01:47:57.534630+00:00 app[web.1]: at Function.Module._load (node:internal/modules/cjs/loader:778:27) 2022-03-29T01:47:57.534630+00:00 app[web.1]: at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) 2022-03-29T01:47:57.534631+00:00 app[web.1]: at node:internal/main/run_main_module:17:47 { 2022-03-29T01:47:57.534631+00:00 app[web.1]: code: 'MODULE_NOT_FOUND', 2022-03-29T01:47:57.534632+00:00 app[web.1]: requireStack: [] 2022-03-29T01:47:57.534632+00:00 app[web.1]: }

@celestinecyr 

I have the node_modules

![image](https://user-images.githubusercontent.com/81438449/160535824-fe2e1963-4c78-4f88-b25a-5f952fa1512f.png)


I never used **Heroku** before, therefore I removed **node_modules** folder completely then and I run **npm install** again. After that I **run npm start** which was working.

**Comment by jonasnapoles** [comment]
I hope I helped

**Comment by jonasnapoles** [comment]
**SOLVED**

**Comment by Adrian-opo** [comment]
![image](https://user-images.githubusercontent.com/85192195/161773340-a736099e-c0e3-442a-afab-3d3140bd08a9.png)
I had a similar error could anyone help

**Comment by yin130** [comment]
Fixed by deleting node_modules directory, yarn.lock, package-lock.json and then installing again.

**Comment by PriapusZeGod** [comment]
Hey, I solved this by using `npm install axios --save` in my terminal(I am using VSCode and I used the built-in terminal there), hope that solves your problems as well!

**Comment by AviLevin** [comment]
fixing project path like duplicates names in folders or keyboard symbols solv my same issue here

**Comment by Rolarg29** [comment]
> I have the same problem

In my exact same case the problem was that I wasn´t inside the EXACT directory of the file, but in a folder containing each project folder (/Projects)>(Projects/project1)> node myproject1.js 
I hope this helps :)

**Comment by daryazata** [comment]
@jonasnapoles how is it solved? For me still an issue

**Comment by BugzTheBunny** [comment]
Yo, Maybe it will help for some of you.

Here is what worked for me:
So, my main folder was named incorrectly, i had some spaces and dots in the root folder name, so I removed them, and i restarted the VSCode.

After that, for some reason `npm run build` did work for me.

and my package.json:
{
  "name": "webpack-typescript",
  "version": "1.0.0",
  "description": "Webpack is a bundling tool, it helps with the workflow.\r It helps to take all of the code and source from your code, and create an optimazied version of it, that is ready for distrebution.",
  "main": "index.js",
  "scripts": {
    "build": "webpack",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "ts-loader": "^9.3.0",
    "typescript": "^4.6.4",
    "webpack": "^5.72.0",
    "webpack-cli": "^4.9.2"
  }
}

**Comment by kalaiselvi-ui** [comment]
> Well it seems this method is not working even for you(Even I'm getting the same error).
> 
> We'll figure out something.

for me also not working kindly give suggestions

**Comment by kalaiselvi-ui** [comment]
> **SOLVED**

how did you solved, i did everything deleted node modules and installed again still i am facing the issue kindly suggest me please

**Comment by Hassan-abdellah** [comment]
try deleting the node modules folder and  move the whole project folder to another place in your system and run 
`npm install `

this method solved the problem for me

**Comment by kalaiselvi-ui** [comment]
I will try this one thanks for your suggestion

**Comment by Leoporto07** [comment]
> I thought I had the same problem... and only after drinking a cup of coffee I noticed my mistake: Instead of `npm install` I had typed `node install`... and of course no such module could be found. From the initial error description this looks very similar.

Man I just figure out the problem! Using your comment, it really worked for me, the issue is acctually in every new folder you creat, you have to "install" the module! So I had this problem whe changed my folder, so after I typed `npm install` all has been solved!

**Comment by GabrielCHP** [comment]
GUYS I FOUND THE SOLUTION! Please let the name of your PNG images short, with no SPACE OR CAPITALIZED LETTERS! Try using short names for accessories, body, or whatever you are doing on NFTs! Don't put any special character!

**Comment by Harrilee** [comment]
> GUYS I FOUND THE SOLUTION! Please let the name of your PNG images short, with no SPACE OR CAPITALIZED LETTERS! Try using short names for accessories, body, or whatever you are doing on NFTs! Don't put any special character!

I don't think this is a solution though. I think a proper solution must specify what's the max length limit for all assets, and perhaps solve this error.

**Comment by Harrilee** [comment]
I found out that it is the problem of my node.js version. After a switch from `18.2.0` back to `10.24.1` the problem is solved.

**Comment by Adrian-opo** [comment]
> > > 
> > 
> > 
> > Yeah I'm facing similar problem. ![image](https://user-images.githubusercontent.com/79012023/154729340-5e2663fd-60cd-4f4c-9753-e13ce5fe7a85.png)
> > Any solution(s) for this?
> 
> just check the path is correct in app.js
> 
> const date = require(_**__dirname + "/views/date.js"**_);

no meu caso o  que resolveu foi ir para o diretório correto

**Comment by marxGourav** [comment]
Send me your error so  know your problem more clearly

**Comment by marxGourav** [comment]
> 

Check the dir. Name of respective ejsfile

**Comment by marcuspassos013** [comment]
Esse erro ocorre quando tentamos startar o servidor em alguma pasta diferente na qual os módulos deveriam estar instalados.

**Comment by marcuspassos013** [comment]
This error occurs when the server starts when we try something different where the errors would occur in some different mass.

**Comment by marcuspassos013** [comment]
this error occurs when the server is started in folder outside the project

**Comment by AndNijaz** [comment]
I am making a react project and faced the same error.
![image](https://user-images.githubusercontent.com/77886732/172142994-6d0968eb-e9c3-46e5-8ca6-5071de45479d.png)

**Comment by AndNijaz** [comment]
I found what was problem for me. Actually my folder structure was going like this:
Web Dev & Programming => Udemy ReacJS => -First React Big Project => senbonzakura-food 

And this "&" in the "Web Dev & Progamming" was making the probelm.

**Comment by kumarmeet** [comment]
Just add 
"engines": {
    "node": "16.x",
    "npm": "6.x"
  },

it will work and runs correctly for nodejs REST apis

**Comment by santii-ez** [comment]
In my case i had enter to src folder from the terminal. It was bad standing.

**Comment by sDuyToans** [comment]
I fixed it in the suck ways. First off all i tried to delete modules and package-lock.json then npm i to reinstall it but not work. 
Second i make new forder then delete modules and package-lock.json and remove the remaining items to the new forder, after that npm i to reinstall it but still not work. 
Finnaly i kinda make 1 new file to test and it run on that but not my file i want so i do type ls on my cmd and copy the name of my file from cmd to run node filename.js again and IT WORK!!! i don't know that it really the point when copy the exact name from cdm but i hope the gonna help.

**Comment by kanam9** [comment]
I found a solution (excuse my english)
First, search WHERE is "yournamefile".js then in vs code console do:  cd "the location of yournamefile.js"
finally do: node "yournamefile".js
Technically it's just the location of your file, if it isnt, this is the video that helped me: https://www.youtube.com/watch?v=2FJ7HjW8OjY

**Comment by sagalt** [comment]
I found where my problem comes from. You have to open the cmd and write this:
"cd C:\Users\YourFolder"
"npm init -y"
create a new file index.js
and open cmd write "code ."
this will open vscode and there will be a .json file created 
then in vscode you can write 
node index.js
you will see it works

**Comment by asserpc** [comment]
Saludos 
la solución para mi fue eliminar la carpera node_modules, instalar lite-server, agregar el cambio en el json, y agregar un commit.
luego ya funciono todo ok

**Comment by Liorrr** [comment]
I was able to solve the issue for me. of course it won't help everyone since you may have other related issues causing this.
the main issue I've had is that the node were looking for dependencies that I've removed from the project later on.

# To solved the issue:
## 1.  delete the following:
  node_modules folder, and package-lock.json

### Optional: on 2nd computer with similar issue it didn't worked without running: 
  npm cache clean --force 
  AND
  restart of the IDE before running npm install

## 2. than run: npm install and the node should start up without problems.

hope it helped some of you

**Comment by Shivam-Negi** [comment]
> npm install axios --save

broooooo it worked.......
thankssssssssss

**Comment by parthf** [comment]
> > 
> 
> Yeah I'm facing similar problem. ![image](https://user-images.githubusercontent.com/79012023/154729340-5e2663fd-60cd-4f4c-9753-e13ce5fe7a85.png)
> 
> Any solution(s) for this?

**Comment by majaliju** [comment]
Yeah same here

**Comment by Aksharmeet** [comment]
I had the same issue but with yarn, so  i deleted node modules, yarn.lock and .yarnrc and then ran yarn to install it all again.
Now it's working like a charm

**Comment by pbhDev** [comment]
The error if found here "Error: Cannot find module ... "

**Comment by APACHE-0002** [comment]
O que resolveu pra mim: criei o script na package.json: "dev": "sucrase-node src/server.js" , e dei yarn dev, note que coloquei src/server.js, se eu colocasse somente server.js, ele nao encontraria o arquivo e dava o erro

**Comment by ivan1016017** [comment]
I do not agree tha tthis issue should be closed since there was not an accurate answer. I could find the answer to this issue. First you have to remember that you have npm and npx. In this case you have to use `npx ts-node file_name.ts`. **Always keep in mind that Npm is a tool that is used to install packages. On the other hand, Npx is a tool that is used to execute packages.**

**Comment by MdodoIng** [comment]
remove this line 

### **mongoose.set('useFindAndModify', false);**

**Comment by LucianoSBja** [comment]
The problem is that you are not in the correct folder to run the application or the script, you would have to open the terminal where it is located or search for it with VScode and try again

**Comment by lucidethinker** [comment]
> > > > 
> > > 
> > > 
> > > Yeah I'm facing similar problem. ![image](https://user-images.githubusercontent.com/79012023/154729340-5e2663fd-60cd-4f4c-9753-e13ce5fe7a85.png)
> > > Any solution(s) for this?
> > 
> > 
> > just check the path is correct in app.js
> > const date = require(_**__dirname + "/views/date.js"**_);
> 
> no meu caso o que resolveu foi ir para o diretório correto

DO RESTART, it 

> > 
> 
> Yeah I'm facing similar problem. ![image](https://user-images.githubusercontent.com/79012023/154729340-5e2663fd-60cd-4f4c-9753-e13ce5fe7a85.png)
> 
> Any solution(s) for this?

restart ur PC, it will work fine

**Comment by devvsakib** [comment]
Thats not gonna work at all, try to recreate your app instead, lmao!!

**Comment by igordavid19** [comment]
Olá! Verifica na sua pasta packjson se o caminho realmente está correto para iniciar, da um npm install -g nodemon. Deu certo aqui.

**Comment by fausantosdev** [comment]
> > GUYS I FOUND THE SOLUTION! Please let the name of your PNG images short, with no SPACE OR CAPITALIZED LETTERS! Try using short names for accessories, body, or whatever you are doing on NFTs! Don't put any special character!
> 
> I don't think this is a solution though. A proper solution must specify what's the max length limit for all assets, and perhaps solve this error.

THAT'S JUST THAT, PAY ATTENTION TO THE PATH NAME TO THE PROJECT, IF POSSIBLE DO NOT USE SPECIAL CHARACTERS OR BLANKS

**Comment by MahmoudMardawi** [comment]
@devvsakib  oh my god! how to be smart like this! I can't imagin! waw incredible

**Comment by devvsakib** [comment]
> 

thanku thanku🤣

**Comment by narender56** [comment]
Deleting the project and re-installing worked for me. Just try if you didn't find any solutions 😅

**Comment by MahmoudMardawi** [comment]
I deleted it again and again on linux and windows ... didn't not work
besided I tried to manipulate the scripts in package.json also didn't work
please check this link :
https://github.com/wbkd/webpack-starter

this is the project for pluralsight course... educative

**Comment by MahmoudMardawi** [comment]
> I deleted it again and again on linux and windows ... didn't not work besided I tried to manipulate the scripts in package.json also didn't work please check this link : https://github.com/wbkd/webpack-starter
> 
> this is the project for pluralsight course... educative

Just clone it and run this command : "npm run dev "
as explained in the course

**Comment by Momosa123** [comment]
Hey everyone,
for me the problem was the name of my folder which had an &.
After deleting this character it worked fine

**Comment by ihsan16110** [comment]
Hello everyone,
In my case, there was a problem with my folder naming which had an & . After removing the character, it worked well for me.

**Comment by Pamelamorim1** [comment]
> ### Versão
> _Sem resposta_
> 
> ### Plataforma
> Windows Powershell
> 
> ### Subsistema
> Powershell
> 
> ### Quais etapas reproduzirão o bug?
> Baixando node.js > Abrindo o Windows Powershell > Selecionando a Pasta > node install. # Não está funcionando!
> 
> ### Com que frequência se reproduz? Existe uma condição necessária?
> _Sem resposta_
> 
> ### Qual é o comportamento esperado?
> _Sem resposta_
> 
> ### O que você vê ao invés disso?
> node:internal/modules/cjs/loader:936 throw err; ^
> 
> Erro: Não é possível encontrar o módulo 'C:__MACOSX\hashlips\install' ←[90m em Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)←[39m ←[90m em Function.Module._load (node:internal/modules/cjs/loader:778:27)←[39m ←[90m em Function.executeUserEntryPoint [como runMain] (node:internal/modules/run_main:81:12)←[39m ←[90m em node :internal/main/run_main_module:17:47←[39m { code: ←[32m'MODULE_NOT_FOUND'←[39m, requireStack: [] }
> 
> ### Informação adicional
> ![INCOMODAR](https://user-images.githubusercontent.com/98562080/151458768-04480bff-46af-4820-96d4-19a6bdb5c771.png)



> mesma merda!
> 
> PS C:\hashlips_art_engine-1.1.1_update> node utils/nftport/uploadFiles.js node:internal/modules/cjs/loader:936 throw err; ^
> 
> Erro: Não é possível encontrar o módulo 'C:\hashlips_art_engine-1.1.1_update\utils\nftport\uploadFiles.js' ?[90m em Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)?[39m ?[90m em Function.Module._load (node:internal/modules/cjs/loader:778:27)?[39m ?[90m em Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12 )?[39m ?[90m em node:internal/main/run_main_module:17:47?[39m { code: ?[32m'MODULE_NOT_FOUND'?[39m, requireStack: [] }

No meu caso consegui resolver o mesmo problema, indo em cima do documento js e com o lado direito do mouse cliquei em Abrir terminal integrado. E voltou a funcionar no meu. Espero ter ajudado.
Essa resolução é com base no VsCode.

**Comment by Pamelamorim1** [comment]
No meu caso consegui resolver o mesmo problema, indo em cima do documento js e com o lado direito do mouse cliquei em Abrir terminal integrado. E voltou a funcionar no meu. Espero ter ajudado.
Essa resolução é com base no VsCode.

**Comment by devvsakib** [comment]
> No meu caso consegui resolver o mesmo problema, indo em cima do documento js e com o lado direito do mouse cliquei em Abrir no terminal integrado. E voltou a funcionar no meu. Espero ter ajudado. Essa resolução é com base no VsCode.

can you write in english please?

**Comment by Pamelamorim1** [comment]
> > No meu caso consegui resolver o mesmo problema, indo em cima do documento js e com o lado direito do mouse cliquei em Abrir no terminal integrado. E voltou a funcionar no meu. Espero ter ajudado. Essa resolução é com base no VsCode.
> 
> can you write in english please?
Yes.
Regarding my case, I managed to solve the same problem, going on top of the js document and with the right side of the mouse, I clicked "Open in the integrated terminal". And it worked on mine again. Hope this helps.
This resolution is based on VsCode.

**Comment by vonshirco** [comment]
> npm run dev

This Worked for me, thanks

**Comment by Benja1512** [comment]
I also have the same problem

![image](https://user-images.githubusercontent.com/103325697/200054937-86af436c-ce69-4f39-9b19-7a0ac4e098f9.png)
 but i just found that my index.js wasnt not in general directory, it should be on there. so the index.js cannot be in another file. directory that shouldnt the general directory where node_module should me 
 look.
 
![image](https://user-images.githubusercontent.com/103325697/200055410-1935edaa-37b8-4458-be0e-2c73a29e087d.png)

**Comment by VadymVashchuk** [comment]
Guys I had exactly the same problem and I managed to solve it in my case. When I wrote in VS Code terminal "node script.js" - I had this error. BUT! When I just wrote there "**node ./script.js**" it perfectly worked. 
In other case I had to wrote full path from main folder like this: "**node ./parentfolder/script.js**" - and it also worked. 
Maybe I didn't understand your problem guys, but I fixed my similar case with this simple code change.

**Comment by 123je** [comment]
> También tengo el mismo problema
> 
> ![imagen](https://user-images.githubusercontent.com/103325697/200054937-86af436c-ce69-4f39-9b19-7a0ac4e098f9.png) pero acabo de descubrir que mi index.js no estaba en el directorio general, debería estar allí. por lo que index.js no puede estar en otro archivo. directorio que no debería ser el directorio general donde debería estar node_module. El Index.js no debe estar en el archivo de vista o en el archivo público o en los archivos de rutas, porque cuando ejecute su npm run dev, aparecerá este error común,
> 
> Mira.
> 
> Espero haber solucionado sus problemas colegas. si es así solo comentame.
> 
> ![imagen](https://user-images.githubusercontent.com/103325697/200055410-1935edaa-37b8-4458-be0e-2c73a29e087d.png)
> 
> así que para mí funcionó ![imagen](https://user-images.githubusercontent.com/103325697/200056011-9b3f8684-93b0-4c91-bfb9-8b84a6f62ccb.png)

hola bro creo que estamos en el mismo proyecto mas o menos pero no te entiendo bien como lo solucionastes mira!, pues yo lo que hize fue sacar el archivo index.js y el app pero nada el mismo error 

![image](https://user-images.githubusercontent.com/73980367/201265375-4a41fd41-fb6c-4d1e-96ea-f1786b0e22c6.png)

**Comment by devvsakib** [comment]
@123je  that's not  the accurate solution i guess, cz mine didn't worked

**Comment by saadrupai97** [comment]
Ran into a similar issue with nodemon running through docker,

it'd be worth checking that your "main" file in your package.json is configured to point at the correct entry point

in package.json

"main": "server.js",
"scripts": {
    "start":"nodemon src/server.js",
    "docker:build": "docker build -f ./docker/Dockerfile . "
  },

**Comment by Dipinlal** [comment]
just copy npm folder from where nodejs installed to your project folder. it worked for me.

**Comment by torquan** [comment]
Another possible reason:

I had a file called `.yarnrc.yml` in my project, fixing the required version:
```
yarnPath: .yarn/releases/yarn-3.2.4.cjs
```

On my server, this version was not installed, resulting in a similar error.

**Comment by leandroArraes** [comment]
No terminal chamar o arquivo --> node arquivo.js
eu estou reinstalndo a versão LTS do node para testar

**Comment by mohsenAero** [comment]
the solution is pretty easy. in windows go to the `C:\Users\yourusername\AppData\Roaming` find the npm folder and delete it. then open PowerShell in your project folder and type `npm install` ......

**Comment by CRLL19-81** [comment]
I recommend deleting the folder: node_modules and then running the command: npm install, that works.

**Comment by Executioner47** [comment]
> Thanks that helped

**Comment by Alina-github** [comment]
Reinstalling node modules right in the directory specified in the Error message helped in my case!
--Works now--

**Comment by Nermeen-M** [comment]
I faced the same problem, the reason was that project path contains the letter "&"
Ex: D:\Projects & Assignments\project-1
The solution is to remove or replace this special character, and it works fine with me now.

**Comment by e1800930** [comment]
> ### Version
> 
> _No response_
> ### Platform
> 
> Windows Powershell
> ### Subsystem
> 
> Powershell
> ### What steps will reproduce the bug?
> 
> Downloading node.js > Opening the Windows Powershell > Selecting the Folder > node install. # Not working!
> ### How often does it reproduce? Is there a required condition?
> 
> _No response_
> ### What is the expected behavior?
> 
> _No response_
> ### What do you see instead?
> 
> node:internal/modules/cjs/loader:936 throw err; ^
> 
> Error: Cannot find module 'C:__MACOSX\hashlips\install' ←[90m at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)←[39m ←[90m at Function.Module._load (node:internal/modules/cjs/loader:778:27)←[39m ←[90m at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)←[39m ←[90m at node:internal/main/run_main_module:17:47←[39m { code: ←[32m'MODULE_NOT_FOUND'←[39m, requireStack: [] }
> ### Additional information
> 
> ![BUG](https://user-images.githubusercontent.com/98562080/151458768-04480bff-46af-4820-96d4-19a6bdb5c771.png)

Problem solved with :
Step 1 : rm package-lock.json
Step 2: rm -rf node_module
Step 3: npm install

**Comment by fadsel** [comment]
Remove the corrupted your local copy

```
yum remove nodejs npm
yum clean all
```

Install a fresh one 

```
yum install -y nodejs npm
```
![image](https://user-images.githubusercontent.com/4649112/230884017-d266482a-a57f-4a74-a88f-a7af5c7cb808.png)

You are now good to go 👍

**Comment by mihailshikov** [comment]
> Hello everyone, In my case, there was a problem with my folder naming which had an & . After removing the character, it worked well for me.

this helps for me

**Comment by Astitva877** [comment]
> ### Version
> _No response_
> 
> ### Platform
> Windows Powershell
> 
> ### Subsystem
> Powershell
> 
> ### What steps will reproduce the bug?
> Downloading node.js > Opening the Windows Powershell > Selecting the Folder > node install. # Not working!
> 
> ### How often does it reproduce? Is there a required condition?
> _No response_
> 
> ### What is the expected behavior?
> _No response_
> 
> ### What do you see instead?
> node:internal/modules/cjs/loader:936 throw err; ^
> 
> Error: Cannot find module 'C:__MACOSX\hashlips\install' ←[90m at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)←[39m ←[90m at Function.Module._load (node:internal/modules/cjs/loader:778:27)←[39m ←[90m at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)←[39m ←[90m at node:internal/main/run_main_module:17:47←[39m { code: ←[32m'MODULE_NOT_FOUND'←[39m, requireStack: [] }
> 
> ### Additional information
> ![BUG](https://user-images.githubusercontent.com/98562080/151458768-04480bff-46af-4820-96d4-19a6bdb5c771.png)

To solve this issue. 

- First delete node_modules
- Then delete the package-lock.json file
- Then run npm install
- Then run npx react-native start

**Comment by pterpstra** [comment]
In Kali i had to run 'npm i'. 
After that 'npm audit fix' twice.

**Comment by KryptonBD** [comment]
Anyone who's still looking for a fix running git clean worked for me.
`git clean -fdx `

**Comment by lavisar** [comment]
> Hey everyone, for me the problem was the name of my folder which had an &. After deleting this character it worked fine

TY so much sir! I have spent 2 hours for this problem until I found your answer :) 

p/s: just make sure your parents folder name dont have the "&"

**Comment by fwjyyds** [comment]
hey guys,my problem happened when i install some package but throw error,maybe it work if you install the package successfully.

**Comment by jamidwyer** [comment]
> Anyone who's still looking for a fix running git clean worked for me.

`git clean -fdx` 

This one worked for me! I had been experimenting with different build setups and I think removing a stray cache likely did the trick. Thank you thank you thank you!

**Comment by PherbCampton** [comment]
Delete the node_modules folder.
and then run `npm install ` from powershell...

Worked smoothly for me.

**Comment by Ohnotech** [comment]
I had the same error using 16.13.0.
I tried to remove `node_modules` and install it again several times but didn't work properly.

Fortunately, I tried that practice again with new version * 21.4.0 (Currently using 64-bit executable) and it works well.

**Comment by oliviermartin** [comment]
I had the same issue on Linux and MacOS after migrating from `npm` to `yarn` for a React Native project.
Installing (and enabling) [`corepack`](https://yarnpkg.com/corepack).

1. To install: `npm install -g corepack` or `brew install corepack` (I had to call `brew unlink corepack`)
2. To enable `corepack enable`

In case you have this new error:

```
Internal Error: The "yarn-path" option has been set (in /Users/MyUser/myproject/.yarnrc.yml), but the specified location doesn't exist (/Users/MyUser/myproject/.yarn/releases/yarn-3.6.4.cjs).
    at i (/Users/MyUser/.cache/node/corepack/v1/yarn/3.6.4/yarn.js:446:11180)
```

Removing `/Users/MyUser/myproject/.yarnrc.yml` fixed the new issue.

**Comment by Musarrat-Jahan** [comment]
PS D:\mern-bootstrap> cd backend
PS D:\mern-bootstrap\backend> npm start

> backend@1.0.0 start
> nodemon index.js   

[nodemon] 3.1.4
[nodemon] to restart at any time, enter `rs`  
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,cjs,json
[nodemon] starting `node index.js`
DB_USERNAME: MUSARRATJAHAN
DB_PASSWORD: 85074
server is running successfully on PORT 8000
(node:8152) [MONGODB DRIVER] Warning: useNewUrlParser is a deprecated option: useNewUrlParser has no effect since Node.js Driver version 4.0.0 and will be removed in the next major version
(Use `node --trace-warnings ...` to show where the warning was created)   
error while connecting with the database MongoServerError: bad auth : Authentication failed.
    at Connection.sendCommand (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\connection.js:297:27)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Connection.command (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\connection.js:325:26)
    at async executeScram (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\auth\scram.js:79:22)
    at async ScramSHA1.auth (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\auth\scram.js:39:16)
    at async performInitialHandshake (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\connect.js:101:13)
    at async connect (D:\mern-bootstrap\backend\node_modules\mongodb\lib\cmap\connect.js:19:9) {
  errorResponse: {
    ok: 0,
    errmsg: 'bad auth : Authentication failed.',
    code: 8000,
    codeName: 'AtlasError'
  },
  ok: 0,
  code: 8000,
  codeName: 'AtlasError',
  connectionGeneration: 0,
  [Symbol(errorLabels)]: Set(2) { 'HandshakeError', 'ResetPool' }
}
^S
we are face mongodb connection

**Comment by Musarrat-Jahan** [comment]
plz guide me

**Comment by DeepSeaCatz** [comment]
Geez what a long chain of useless info for 2 years lol

Finally fixed my issue. I was using nvm to install the node v18.20.5 and I realized I wasn't using the latest version of nvm. After I've updated my nvm and re-installed the node version with `nvm uninstall 18.20.5` and `nvm install 18.20.5`, it finally had all the correct files for that version. Close cmd and re-open cmd and npm worked.

You can get the latest nvm here: [https://github.com/coreybutler/nvm-windows/releases](https://github.com/coreybutler/nvm-windows/releases)

**Comment by uninformed816** [comment]
@shabbir23ah https://stackdev.space/questions/what-are-the-benefits-of-incorporating-mongoose-into-the-mean-stack
